cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/AbstractSystemMetadataWriter.java

```diff
@@ -60,6 +62,14 @@ public AbstractSystemMetadataWriter(MetadataStore metadataStore, Id.NamespacedId
   abstract String[] getSystemTagsToAdd();
 
   /**
+   * Define the {@link MetadataScope#SYSTEM system} schema to add for this entity.
+   *
+   * @return the schema as a {@link String}
+   */
+  @Nullable
+  abstract String getSchemaToAdd();
```

Comment:
Since this method can return null, give a default implementation that returns null. This will prevent derived classes that don't return schema from implementing this method.